controllers/hypershiftlogforwarder/hypershiftlogforwarder_controller.go

```diff
@@ -120,23 +118,55 @@ func (r *HyperShiftLogForwarderReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
-	deletion := false
-	// HyperShiftLogForwarder was deleted by CU
-	if !instance.DeletionTimestamp.IsZero() {
-		deletion = true
+	if !instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is being deleted
 		if controllerutil.ContainsFinalizer(instance, constants.ManagedLoggingFinalizer) {
-			controllerutil.RemoveFinalizer(instance, constants.ManagedLoggingFinalizer)
-			if err = r.Update(ctx, instance); err != nil {
+			hlfDeletion = true
+
+			//delete the clf if present
+			if clfFound {
+				err = r.MCClient.Delete(ctx, clf)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+			}
+
+			//create new clf from clft
+			if err := r.createClusterLogForwarderFromTemplate(ctx, req); err != nil {
```

Comment:
why we call the create from CLFT in this controller? it will make the two existing controllers a bit mess?

Reply:
fixed